extensions/kafka/deployment/src/main/java/org/apache/camel/quarkus/component/kafka/deployment/KafkaProcessor.java

```diff
@@ -68,4 +69,11 @@ public void configureKafkaComponentForDevServices(
             }
         }
     }
+
+    @BuildStep
+    public ReflectiveClassBuildItem reflectiveClasses() {
+        return new ReflectiveClassBuildItem(true, true,
+                "org.apache.kafka.clients.producer.KafkaProducer",
+                "org.apache.kafka.clients.consumer.KafkaConsumer");
+    }
```

Comment:
Also, I wonder whether the class `org.apache.kafka.clients.producer.internals.Sender` would need [to be registered also](https://github.com/apache/camel/blob/main/components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java#L111..L115)

At the end of the day, maybe adding testing covering each usage of ReflectionHelper in `KafkaFetchRecords` and `KafkaProducer` would be helpful ?

Reply:
yeah, ```org.apache.kafka.clients.producer.internals.Sender``` should be added here.